src/components/Api.js

```diff
@@ -0,0 +1,107 @@
+export default class Api {
+  constructor(options) {
+    this.baseUrl = options.baseUrl;
+    this.headers = options.headers;
+    //this.cardId = cardElement.cardId;
+
+  }
+
+checkResponse(res) {
+  if (res.ok) {
+    return res.json();
+  }
+  return Promise.reject(`Error: ${res.status}`);
+}
+
+  //USER ROUTES
+
+  getUserInfo() {
+    return fetch(`${this.baseUrl}/users/me`, {  
+      headers: this.headers,
+    }).then(this.checkResponse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this.checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
![Errors of note](https://github.com/user-attachments/assets/5d0b86ed-2964-4295-8733-f52f5809ff93)
![Errors of note](https://github.com/user-attachments/assets/2e20f08a-5548-4c4e-babb-64c732335497)

Reply:
Hi! thank you so much for this suggestion. I understand it conceptually... but perhaps I applied it wrong. Is the maximum call stack size a processing error on my end or a kind of "overload" of sorts?

Reply:
Hi)

You `_request` should be like this. You can copy it

```js


  _request(url, options) {
    return fetch(url, options).then(this.checkResponse)
  }

```

Reply:
Ok I have copied it to double check. And tested one by one. After two were successful I applied it to all of my api calls. And received the same response. So from what I think I'm understanding is that the syntax is correct but there are too many repetitive requests on the call stack? does this mean I should refactor it some how more?
![Errors of note](https://github.com/user-attachments/assets/24355899-a59d-466e-91be-f564bb529741)
![Errors of note](https://github.com/user-attachments/assets/6dcf43a6-1161-4c30-9d9c-dc2a1c7ca151)